service/history/replicatorQueueProcessor.go

```diff
@@ -494,10 +494,6 @@ func (p *replicatorQueueProcessorImpl) getVersionHistoryItems(
 ) ([]*historyspb.VersionHistoryItem, []byte, error) {
 
 	versionHistories := mutableState.GetExecutionInfo().GetVersionHistories()
-	if versionHistories == nil {
```

Comment:
this is not possible any more?

Reply:
version history is set since [v1.0.0](https://github.com/temporalio/temporal/releases/tag/v1.0.0)

`newMutableStateBuilderWithVersionHistories` is used for [creation of new workflow](https://github.com/temporalio/temporal/blob/8d8720b55fb7b3ccc8a9e27d33bb1b3f2da69fb5/service/history/historyEngine.go#L475)

`newMutableStateBuilder` is not used for initialization
```bash
temporal|8d8720b55 ⇒ rg 'newMutableStateBuilder\(' 
service/history/mutableStateBuilder_test.go
108:	s.msBuilder = newMutableStateBuilder(s.mockShard, s.mockEventsCache, s.logger, testLocalNamespaceEntry)

service/history/historyBuilder_test.go
105:	s.msBuilder = newMutableStateBuilder(s.mockShard, s.mockEventsCache,

service/history/workflowExecutionContext.go
245:		c.mutableState = newMutableStateBuilder(
318:		c.mutableState = newMutableStateBuilder(

service/history/mutableStateBuilder.go
174:func newMutableStateBuilder(
253:	s := newMutableStateBuilder(shard, eventsCache, logger, namespaceEntry)
temporal|8d8720b55
```

Reply:
plus backfill when [loading](https://github.com/temporalio/temporal/blob/v1.3.0/service/history/mutableStateBuilder.go#L307-L320) ms